### PR TITLE
LPS-176212 Error message is not displayed when a user views an expired kb article

### PR DESCRIPTION
**Pull request by marcogalluzzi**
Expired KB articles should only be visible by those users with UPDATE permission.

I only have a doubt: should expired articles be listed?

# Step to reproduce:

1. Enable FF LPS-165476
2. Add two kb articles and expire one of them
3. Add a regular role with these permissions (in Knowledge Base):
   1. GENERAL PERMISSIONS: Access in Site and Asset Library Administration
   2. KNOWLEDGE BASE ARTICLE: View
4. Add a new user and give her the regular role
5. Login with the new user
6. Go to KB admin. See how both expired and non expired articles are listed Captura de pantalla 2023-03-06 a las 11 15 07
8. Click the expired article title

*Expected Results:*
An error message is displayed. See: Captura de pantalla 2023-03-06 a las 11 15 19

*Current Results:*
No error message is displayed.

### CODE REVIEW

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
ci:test:relevant

**Comment by marcogalluzzi** [comment]
ci:test:relevant

**Review by AliciaGarciaGarcia — COMMENTED** [review]
Q! 

Hi @marcogalluzzi 

I'm not sure about this solution.

I read the slack messages and I do not think that this is the correct behavior.

The user has the VIEW permission, and we let the user to click on it to VIEW it, and I do not know if throw an error message on the action we let her do it's good idea.

> as said, from my perspective we could not reveal the existence of this article if an admin or the owner expired this

If we can not see it, why list it?

I check how other services are doing it, and on Web Content it let you list and preview it (our view) 

https://user-images.githubusercontent.com/47524751/223164517-d2d6639a-0d90-4924-88bc-d619d872cacb.mp4

I think that it breaks the way other places works :(

If this should be this way, maybe do not put the link to view the article is more clear?

On Documents and media, it does not show the expired version

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/security/permission/resource/KBArticleModelResourcePermissionWrapper.java [hunk 1]
```diff
@@ -63,6 +63,20 @@
...
+				consumer.accept(
+					(permissionChecker, name, kbArticle, actionId) -> {
+						if (kbArticle.isExpired() &&
+							!permissionChecker.hasPermission(
+								kbArticle.getGroupId(), name,
+								kbArticle.getResourcePrimKey(),
+								ActionKeys.UPDATE)) {
+
+							return false;
+						}
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/security/permission/resource/KBArticleModelResourcePermissionWrapper.java:75]: This should only apply to the VIEW action:
```
if (actionId.equals(ActionKeys.VIEW) && kbArticle.isExpired() ...)
```

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Pull request closed without merging**